rama-http/src/layer/auth/require_authorization.rs

```diff
@@ -183,8 +213,14 @@ where
         request: Request<B>,
     ) -> Result<(Context<S>, Request<B>), Response<Self::ResponseBody>> {
         match request.headers().get(header::AUTHORIZATION) {
-            Some(actual) if actual == self.header_value => Ok((ctx, request)),
+            Some(actual) if actual == self.credential.header_value => Ok((ctx, request)),
```

Comment:
It's a bit philosophical, but I consider this not correct implementation. Anonymous users should be anonymous because the header is not defined. IMHO if the header is defined it has to be correct. In Proxy-Auth you do that correct, but here in this file you also are allowing it to be anonymous if credential is wrong, which is not correct in my opinion.